src/lib/core/core.gni

```diff
@@ -13,6 +13,11 @@
 # limitations under the License.
 
 declare_args() {
+  # Enable recommended settings by default. This is a meta-setting
+  # that acts as a default for various other settings. Setting it
+  # to false produces a more conservative / minimal set of defaults.
+  matter_enable_recommended = true
```

Comment:
The name is reasonably generic to be unclear about what it actually enables/disables (and this seems to be the intent: a setting that is vague on what it actually enables because in the future it seems to be intended to enable more and more).

I would maybe document the intent of guarding against new features that may increase size of the binary. We also need a clear `why` rather than just what it does. The PR summary also could use some work here: in what circumstances is this to be used? Is it just decreasing binary size or something else?

Reply:
I wouldn't really look at it as "enabling more and more" in the future, but rather being a single setting that can be disabled (since it's on by default) to get a more conservative set of default options. This is particularly relevant when building for Linux (and to a lesser extent Darwin), because we seem to often use `current_os == "Linux"` to enable various "debuggy" or optional things that we don't enable by default when building for other platforms. E.g.

- `chip_use_data_model_interface` defaults to `check` on Linux but `disabled` on other platforms
- `chip_enable_transport_trace` is on by default on Linux and Darwin and disabled otherwise

Maybe another way to approach to this is to ask, what is our approach within the SDK when adding features in terms of deciding whether they should be on by default?

Reply:
I’ve added some more comments about when this could / should be used.